src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/ViolationTicketMapper.java

```diff
@@ -197,6 +200,30 @@ default ca.bc.gov.open.jag.tco.oracledataapi.api.model.DisputeCount mapDisputeCo
 				return count;
 			}
 		}
-	    return null;
+		return null;
+	}
+
+	@AfterMapping
+	default void setLawyerAddress(@MappingTarget ViolationTicket violationTicket, Dispute dispute) {
+		if (dispute != null && violationTicket != null && violationTicket.getDispute() != null && !StringUtils.isBlank(dispute.getLawyerAddress())) {
+			int addressLength = dispute.getLawyerAddress().length();
+			String lawyerAddress = dispute.getLawyerAddress();
+			String addressLine1 = "";
+			String addressLine2 = "";
+			String addressLine3 = "";
+			if (addressLength > 200) {
+				addressLine1 = lawyerAddress.substring(0, 100);
+				addressLine2 = lawyerAddress.substring(100, 200);
+				addressLine3 = lawyerAddress.substring(200, addressLength);
+			} else if (addressLength > 100) {
+				addressLine1 = lawyerAddress.substring(0, 100);
+				addressLine2 = lawyerAddress.substring(100, addressLength);
+			} else {
+				addressLine1 = lawyerAddress;
+			}
+			violationTicket.getDispute().setLawFirmAddrLine1Txt(addressLine1);
+			violationTicket.getDispute().setLawFirmAddrLine2Txt(addressLine2);
+			violationTicket.getDispute().setLawFirmAddrLine3Txt(addressLine3);
+		}
```

Comment:
Can you write a junit test for this? Confirm address line is constructed as excepted?

Reply:
Yeah for sure. Thanks for reminding that.